pkg/syndesis/action/checkupdates.go

```diff
@@ -0,0 +1,49 @@
+package action
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	"github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/syndesis/version"
+)
+
+// Checks if the syndesis installation should be upgraded and move to the "Upgrading" status.
+type CheckUpdates struct {
+	operatorVersion	string
+}
+
+func (a *CheckUpdates) CanExecute(syndesis *v1alpha1.Syndesis) bool {
+	return syndesisInstallationStatusIs(syndesis, v1alpha1.SyndesisInstallationStatusInstalled)
```

Comment:
Is it really that we can updates only when syndesis is in "Installed" ? What's about "StartupFailed" ?

Actually, my question is probably more what about todo when Syndesis is in an error state like `StartupFailed`. How can we get back to a 'good' state ? An update here could be an option or a 'force' reinstall.

Reply:
Yes, I'll add `StartupFailed` as other option. We can move on from `StartupFailed` if we put syndesis in a situation where all required pods are running, the switch happens automatically. But doing a upgrade/downgrade is also a possible option for cases where pods don't start because of a software error.